tests/app/celery/test_process_delivery_status_result_tasks.py

```diff
@@ -17,196 +26,358 @@ def sample_delivery_status_result_message():
-    mocker,
-    notify_db_session,
-    sample_delivery_status_result_message,
-    sample_translate_return_value,
-    sample_notification,
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks.process_delivery_status(event=sample_delivery_status_result_message)
+
+
+# test get_provider_info when no provider is given by the celery event
+def test_get_provider_info_with_no_provider(
+        notify_db_session,
+        sample_notification_platform_status,
+        sample_sqs_message_without_provider
 ):
-    """Test that celery will retry the task if provider doesnt exist then self.retry is called"""
+    mock_celery_task = MockCeleryTask()
 
-    # change message['provider'] to invalid provider name
-    sample_delivery_status_result_message["message"]["provider"] = "abc"
-    mocker.patch("app.clients")
-    mocker.patch("app.clients.sms.SmsClient")
-    mocker.patch(
-        "app.clients.sms.twilio.TwilioSMSClient.translate_delivery_status",
-        return_value=sample_translate_return_value,
+    # now supply the sample to the function we want to test
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks._get_provider_info(mock_celery_task, sample_sqs_message_without_provider)
+
+
+def test_get_provider_info_with_invalid_provider(
+        notify_db_session,
+        sample_sqs_message
+):
+    """Test that _get_provider_info() will raise a celery retry when sqs message has an invalid provider"""
+
+    # mock the celery task event
+    mock_celery_task = MockCeleryTask()
+    sample_sqs_message['provider'] = "abc"
+
+    # now supply the sample to the function we want to test with the expectation of failure
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks._get_provider_info(mock_celery_task, sample_sqs_message)
+
+
+# here we test all valid provider names
+@pytest.mark.parametrize('provider_name', ['twilio'])
+def test_get_provider_info_with_valid_provider(
+        notify_db_session,
+        sample_notification_platform_status,
+        sample_sqs_message,
+        provider_name
+):
+
+    """Test that _get_provider_info() will return  a celery retry when sqs message has an invalid provider"""
+    # default provider_name to current parameterized value
+    sample_sqs_message['provider'] = provider_name
+
+    # mock the celery task event
+    mock_celery_task = MockCeleryTask()
+
+    # now supply the sample to the function we want to test
+    provider_name_output, provider = process_delivery_status_result_tasks._get_provider_info(
+        mock_celery_task,
+        sample_sqs_message
     )
-    mocker.patch(
-        "app.celery.process_delivery_status_result_tasks.attempt_to_get_notification",
-        return_value=(sample_notification, False, False),
+
+    # parameterized provider_name should match the output from _get_provider_info
+    # and it should also match the provider.name
+    assert provider_name == provider_name_output == provider.name
```

Comment:
Hmm.  At a minimum, this is inconsistent with Pep8.  Scroll up from [here](https://pep8.org/#function-annotations).

If `provider_name == provider_name_output` is true, this becomes an assertion that `True == provider.name`.